src/client/datascience/notebook/helpers/executionHelpers.ts

```diff
@@ -97,6 +98,34 @@ export async function updateCellWithErrorStatus(
     });
 }
 
+// Update the code contents of the cell
+export async function updateCellCode(cell: NotebookCell, text: string) {
+    // Use Workspace edit to apply a replace to the full cell text
+    const edit = new WorkspaceEdit();
+    edit.replace(
+        cell.document.uri,
+        new Range(cell.document.lineAt(0).range.start, cell.document.lineAt(cell.document.lineCount - 1).range.end),
+        text
+    );
+    await workspace.applyEdit(edit);
```

Comment:
Quick note here. We used to guard against direct access to vscode classes using our injected interfaces. But with native notebooks and our .vscode tests this doesn't seem as important to me. I think that there is access to workspace edit off of IDocumentManager so I could use that, but to me it doesn't seem needed here.

Reply:
But open to the idea if we think we should still do this.

Reply:
I think this is perfectly fine, fewer interfaces/dependencies injected makes the code easier to manage, at least IMHO.